base/operators.jl

```diff
@@ -822,6 +822,17 @@ julia> [1:5;] |> x->x.^2 |> sum |> inv
 """
 |>(x, f) = f(x)
 
+"""
+    |>>(x, f)
+
+Applies a function to the preceding argument, and returns that argument.
+This allows for easy tee function chaining.
```

Comment:
Is "tee function chaining" a popular terminology for this?

Also, I guess it'd be nice to have some motivating examples as doctests. Is it for mutating functions like `pop!` and side-effects like `display`?

Reply:
> Is "tee function chaining" a popular terminology for this?

I have no idea :) I think "tee" is used for this, but if the general idea here gets accepted, we should revisit the terminology.

The main use case I've found is file saving in queryverse:
```julia
df |> @filter(...) |>> save("foo.csv") |> @vlplot(...) |>> save("foo.pdf")
```
is just something I'd like a lot :) The main use case I have is side-effects. Your display example is another one.

Reply:
FYI, Python has [`itertools.tee`](https://docs.python.org/3/library/itertools.html#itertools.tee) which has a different meaning (especially when the input is a "stateful" iterator).  That's why I wondered if it has a standard meaning. But clarifying the terminology after this is accepted makes sense.

I think `save` example is appealing. I guess an objection may be that it can already be done if `FileIO.save` returns the second argument.  But I suppose there are cases where this is not possible.

Reply:
there is also https://en.wikipedia.org/wiki/Tee_(command)

Reply:
Yes, that is the one where I got the name from!

Reply:
The `tee` command is closer to Python's `itertools.tee` than `|>>`.

I guess the question is if it is "allowed" to specialize `|>>(x, f)` in case `x` is some kind of "lazy" object (e.g., iterators) to minimize some overheads.  For example, in principle, it is possible to run filtering in `@filter(...) |>> save("foo.csv") |> @vlplot(...)` only once.  It would then be closer to the `tee` command.